dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/GoalsController.cs

```diff
@@ -88,7 +111,46 @@ public class GoalsController : Controller
         [ValidateAntiForgeryToken]
         public async Task<ActionResult> RemoveGoal(Guid id, Guid personId, Guid recordId)
         {
-            await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.DeleteGoalAsync(id.ToString()), personId, recordId);
+            await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.Goals.DeleteAsync(id.ToString()), personId, recordId);
+            return RedirectToAction("Index", new { personId = personId, recordId = recordId });
+        }
+
+        /// <summary>
+        /// Get a goal recommendation for the logged in user
+        /// </summary>
+        [HttpGet]
+        public ActionResult GoalRecommendation(Guid personId, Guid recordId)
+        {
+            var recommendation = new GoalRecommendation
+            {
+                AssociatedGoal = new Goal()
+            };
+            return View(recommendation);
+        }
+
+        /// <summary>
+        /// Create a goal recommendation for the logged in user
+        /// </summary>
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> GoalRecommendation(GoalRecommendation recommendation, Guid personId, Guid recordId)
+        {
+            if (recommendation?.AssociatedGoal?.RecurrenceMetrics?.OccurrenceCount == null && recommendation?.AssociatedGoal?.RecurrenceMetrics?.WindowType == GoalRecurrenceType.Unknown.ToString())
```

Comment:
Maybe. I don't think it hurts to be redundant in this case in case bits are copy/pasted